hyperspy/signal.py

```diff
@@ -4773,10 +4785,12 @@ def copy(self):
             self._plot = backup_plot
 
     def __deepcopy__(self, memo):
-        dc = type(self)(**self._to_dictionary())
+        dc = type(self)(**self._to_dictionary(add_original_metadata=False))
         if isinstance(dc.data, np.ndarray):
             dc.data = dc.data.copy()
 
+        dc.original_metadata = self.original_metadata.copy()
```

Comment:
I find this counterintuitive: I think that users would expect that ``deepcopy`` deepcopies everything. That's documented in the ``deepcopy`` method below, but this may be frustrating for users who fail to notice the exception made with ``original_metadata``.

Reply:
Yes, I thought the same: with deepcopy, we would expect a new copy, however, the [python documentation](https://docs.python.org/3/library/copy.html) implies that not everything is _always_ copied. A more convincing point is that `original_metadata` implies that these metadata are original and therefore not expected to be changed? Maybe the best would be to freeze it.

Reply:
I am actually not very convinced about any of those 2 arguments. I still think that the expected behavior of ``deepcopy`` is deepcopying everything. As for freezing original metadata, the issue is that often the metadata stored using the microscope's software is wrong, so some may like to use HyperSpy to fix the metadata. By not deepcopying it, somebody could crop the data and amend the ``original_metadata`` in the cropped signal unaware that he's modifying the ``original_metadata`` of the original signal.

Reply:
Since this is too contentious, I have reverted these changes and updated the changelog and summary/title of this PR accordingly.